src/test/java/org/springframework/integration/kafka/outbound/OutboundTests.java

```diff
@@ -76,9 +82,42 @@ public static void tearDown() {
 
 	@Test
 	public void testAsyncProducerFlushed() throws Exception {
-		KafkaConsumerContext<String, String> consumerContext = createConsumer();
-		// pre-consume to start the receiver because the high-level API doesn't support --from-beginning
-		consumerContext.receive();
+
+		// create the topic
+
+		try {
+			AdminUtils.createTopic(kafkaRunning.getZkClient(), TOPIC, 1, 1, new Properties());
+		}
+		catch (TopicExistsException e) {
+			// do nothing
+		}
+
+		final String suffix = UUID.randomUUID().toString();
+
+		ZookeeperConfiguration configuration = new ZookeeperConfiguration(new ZookeeperConnect());
+		DefaultConnectionFactory connectionFactory = new DefaultConnectionFactory(configuration);
+		connectionFactory.afterPropertiesSet();
+		final KafkaMessageListenerContainer kafkaMessageListenerContainer = new KafkaMessageListenerContainer(connectionFactory, TOPIC);
```

Comment:
This is a Zookeeper/Kafka problem. Sometimes, on repeated runs, due to the topic deletion in teardown, `createTopic` fails, but the metadata doesn't exist yet. So we need to run on a clean kafka cluster.

I think that the Listener should fail starting if there is no topic to listen to, but we need to make that part more robust (I think it's a NPE now, which is bad).